<launch>

    <!-- Known transform copied from realsense-ros driver in ros2 -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_internal_frame_broadcaster_5" 
      args="0.0 0.0 0.0   0.0 0.0 0.0 1.0 camera_link camera_color_frame" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_internal_frame_broadcaster_6"
      args="0.0 0.0 0.0   -0.5 0.5 -0.5 0.5 camera_color_frame camera_color_optical_frame" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_internal_frame_broadcaster_7"
      args="0.0 0.0 0.0   0.0 0.0 0.0 1.0 camera_link camera_depth_frame" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_internal_frame_broadcaster_8"
      args="0.0 0.0 0.0   -0.5 0.5 -0.5 0.5 camera_depth_frame camera_depth_optical_frame" />

  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="fake_link_transform"
      args="0.0 0.0 0.0   0.0 0.0 1.0 0.0 base_link fake_link" /> -->



  <node type="point_cloud2_assembler" pkg="laser_assembler" name="my_assembler">
    <!-- cloud = topic to recieve clouds on -->
    <remap from="cloud" to="/pcl_stitcher_input_processed"/> 
    <param name="max_clouds" type="int" value="400" />
    <param name="tf_cache_time_secs" type="double" value="45.0" />
    <!-- I think this sets the fixed frame to be base_link, and tf takes care of everything else-->
    <param name="fixed_frame" type="string" value="base_link" />
  </node>

  <node name="realsense_processing" pkg="single_camera_demo" type="realsense_processing.py">
    <param name="pcl_fwd_status" type="boolean" value="false" />
    <param name="fwd_pcd_with_timer" type="boolean" value="false" />
  </node>


  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

  <node pkg="nodelet" type="nodelet" name="cropbox" args="load pcl/CropBox pcl_manager" output="screen">
    <remap from="~input" to="/pcl_stitcher_input_raw" />
    <remap from="~output" to="/pcl_stitcher_input_processed" />
    <rosparam>
      keep_organized: False
      input_frame: /base_link
    </rosparam>
  </node>


</launch>